src/relay_pic/main.c

```diff
@@ -0,0 +1,153 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <xc.h>
+#include <pic16f1826.h>
+#include "mcc_generated_files/adc.h"
+#include "mcc_generated_files/mcc.h"
+
+typedef unsigned char uint8_t;
+typedef unsigned short uint16_t;
+
+#define POWER (1 << 2)
+#define SELECT (1 << 3)
+#define LIM1 (1 << 7)
+#define LIM2 (1 << 6)
+#define DIP1 (1 << 4)
+#define DIP2 (1)
+#define DIP3 (1 << 2)
+#define DIP4 (1 << 3)
+#define LED (1 << 5)
+#define CURR_SENSE1 (1 << 1)
+#define CURR_SENSE2 (1)
+
+uint16_t dipInputs;
+uint16_t adcResult;
+uint16_t i2cSlaveRecv; // Data received on i2c
+uint16_t i2cSlaveSend; // Data to be sent on i2c
+
+void i2cSlaveInit(uint16_t address) {
+    SSPSTAT = 0x80;
+    SSPADD = address;
+    SSPCON = 0x36;
+    SSPCON2 = 0x01;
+    TRISB1 = 1; // SDA
+    TRISB4 = 1; // SCL
+    GIE = 1;
+    PEIE = 1;
+    SSP1IF = 0;
+    SSP1IE = 1;
+}
+
+void interrupt i2cSlaveInt(void) { 

```

Comment:
presumably you're gonna have other interrupts firing. You can only set up one interrupt service routine, so I'd name this to something like `interrupt_service_routine` or something.